helm/templates/core-ingress.yaml

```diff
@@ -0,0 +1,62 @@
+{{- if .Values.ingress.core.enabled -}}
+{{- $fullName := include "substrate-telemetry.fullname" . -}}
+{{- $svcPort := .Values.service.core.port -}}
+{{- if and .Values.ingress.core.className (not (semverCompare ">=1.18-0" .Capabilities.KubeVersion.GitVersion)) }}
+  {{- if not (hasKey .Values.ingress.core.annotations "kubernetes.io/ingress.class") }}
+  {{- $_ := set .Values.ingress.core.annotations "kubernetes.io/ingress.class" .Values.ingress.core.className}}
+  {{- end }}
+{{- end }}
+{{- if semverCompare ">=1.19-0" .Capabilities.KubeVersion.GitVersion -}}
+apiVersion: networking.k8s.io/v1
+{{- else if semverCompare ">=1.14-0" .Capabilities.KubeVersion.GitVersion -}}
+apiVersion: networking.k8s.io/v1beta1
+{{- else -}}
+apiVersion: extensions/v1beta1
+{{- end }}
```

Comment:
Do we need this sort of block, since we know what version of k8s we'll be running (and probably wouldn't "test" the other branches?

Reply:
these are the defaults for the helm chart to make sure that this deployment would be compatible with all the Kubernetes versions so you can just ignore them.
about k8s version, they are configured to be upgraded automatically, so I think it would make sense to keep these configuration here.